openpype/lib/applications.py

```diff
@@ -1649,11 +1649,7 @@ def prepare_context_environments(data, env_group=None, modules_manager=None):
     project_doc = data["project_doc"]
     asset_doc = data["asset_doc"]
     task_name = data["task_name"]
```

Comment:
Since these are now not checked - let's move them closer to where they are used lower down. Also, since `asset` now becomes optional it seems, shouldn't this be `data.get("asset_doc")` and `data.get("task_name")`?

Reply:
The keys are always set.